p2p/src/peer_manager/peer_context.rs

```diff
@@ -45,6 +45,9 @@ pub struct PeerContext<A> {
     /// Sent ping details
     pub sent_ping: Option<SentPing>,
 
+    /// Whether an address list request was sent and no response was received.
```

Comment:
Please explain with more details why this is necessary and how it helps in the lifetime of the application.

Reply:
Sure, updated.